integration/simulation/simulation_tester.go

```diff
@@ -2,25 +2,33 @@ package simulation
 
 import (
 	"math/rand"
+	"runtime"
 	"testing"
 	"time"
 
 	"github.com/obscuronet/obscuro-playground/integration/simulation/network"
 	"github.com/obscuronet/obscuro-playground/integration/simulation/params"
 
+	"github.com/obscuronet/obscuro-playground/go/log"
 	stats2 "github.com/obscuronet/obscuro-playground/integration/simulation/stats"
 
 	"github.com/google/uuid"
 )
 
 // testSimulation encapsulates the shared logic for simulating and testing various types of nodes.
 func testSimulation(t *testing.T, netw network.Network, params *params.SimParams) {
+	defer func() {
+		// wait until clean up is complete before we log the lingering goroutine count
```

Comment:
@BedrockSquirrel Are you sure `defer netw.TearDown()` is called before this `defer`? It's not clear to me it would be. I think you need:

```
defer func() {
        // wait until clean up is complete before we log the lingering goroutine count
	defer log.Info("goroutine leak monitor - simulation end - %d goroutines currently running", runtime.NumGoroutine())
        netw.TearDown()
}()
```

Reply:
@joeldudley yeah defers are guaranteed reverse order (they're put on a LIFO stack https://go.dev/tour/flowcontrol/13 - I guess because the first resource open is often the last to be closed).

So thought it might be simpler to keep it with the other log message and to avoid ordering risk if someone ever added another cleanup defer to the simulation runner. But don't feel strongly, happy to change if you prefer.